src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,226 @@
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                return resultSet.getLong("cart_id");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting shopping cart id by user id "
+                    + userId + " failed. ", e);
+        }
+        return null;
+    }
+
+    private List<Product> getProductToShoppingCartFromDB(Long cartId) {
+        String query = "SELECT shopping_carts_products.product_id "
+                + " FROM shopping_carts_products "
+                + " JOIN products "
+                + " ON shopping_carts_products.product_id = products.product_id "
+                + " WHERE shopping_carts_products.cart_id = ? "
+                + " AND shopping_carts_products.deleted = FALSE ";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cartId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                products.add(getProductFromDbById(resultSet.getLong("product_id")));
+            }
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get products list by cart id "
+                    + cartId, e);
+        }
+    }
+
+    private Product getProductFromDbById(Long productId) {
+        String query = "SELECT name, price FROM products WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, productId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String name = resultSet.getString("name");
+                double price = resultSet.getDouble("price");
+                Product product = new Product(name, price);
+                product.setId(productId);
+                return product;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting product by id "
+                    + productId + " failed. ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean clear(ShoppingCart shoppingCart) {
+        Long cartId = shoppingCart.getId();
+        for (Product product : shoppingCart.getProducts()) {
+            deleteProduct(cartId, product);
+        }
+        shoppingCart.getProducts().clear();
+        return shoppingCart.getProducts().isEmpty();
+    }
```

Comment:
This doesn't look like DAO logic, so maybe this should be a private local method? I think we also had a similar method in the Service layer?